src/zenml/steps/utils.py

```diff
@@ -538,6 +538,7 @@ def single_step_pipeline() -> None:
         __step(**inputs)
 
     run = single_step_pipeline.with_options(unlisted=True)()
+    assert run
```

Comment:
I updated the return type of `Pipeline.__call__` to be an `Optional[PipelineRunResponse]` instead of `Any`. This enables autocomplete on the run object. However, as this is optional (`None` in case of scheduled runs), I need this assert for mypy.